### PR TITLE
Update conversion calc. impl. and test it

### PR DESCRIPTION
**Pull request by 7sne**

### CODE REVIEW

File: pages/eth-unit-conversion.test.tsx [hunk 1]
```diff
@@ -51,4 +51,34 @@ describe(EthUnitConversion.name, () => {
...
+
+    fireEvent.change(gweiField, { target: { value: '140005.54/fa' } })
+    expect(gweiField.value).toEqual('140005.54/fa')
+    expect(weiField.value).toEqual('140005540000000')
+    expect(ethField.value).toEqual('0.00014000554')
+  })
+
+  it('types letters and special signs to one of the fields and error gets displayed', async () => {
+    const root = render(<EthUnitConversion />)
+    const errorField = (await root.findByTestId('error')) as HTMLElement
```
Comment (hasparus) [inline pages/eth-unit-conversion.test.tsx:74]: Do we need this `as HTMLElement` in this case?

File: pages/token-unit-conversion.test.tsx [hunk 2]
```diff
@@ -44,4 +44,30 @@ describe(TokenUnitConversion.name, () => {
...
+    expect(baseField.value).toEqual('0.01244444400000000055')
+
+    fireEvent.change(unitField, { target: { value: '12444444000000000.55/fa' } })
+    expect(unitField.value).toEqual('12444444000000000.55/fa')
+    expect(baseField.value).toEqual('0.01244444400000000055')
+  })
+
+  it('types letters and special signs to one of the fields and error gets displayed', async () => {
+    const root = render(<TokenUnitConversion />)
+    const errorField = (await root.findByTestId('error')) as HTMLElement
```
Comment (hasparus) [inline pages/token-unit-conversion.test.tsx:64]: I'm not quite sure `as HTMLElement` is needed here. Isn't this already `HTMLElement`?

**Review by hasparus — APPROVED** [review]

File: misc/unitSchema.ts [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import { z } from 'zod'
+
+export const unitSchema = z.preprocess(
+  (value) => String(value).trim(),
+  z
+    .string()
+    .regex(/^[.0-9]*$/, { message: "The value mustn't contain letters" })
+    .min(1, { message: 'The value must be minimum 1 numbers long' })
```
Comment (krzkaczor) [inline misc/unitSchema.ts:8]:
```suggestion
    .min(1, { message: 'The value must be bigger or equal to 1' })
```

File: misc/unitSchema.ts [hunk 4]
```diff
@@ -0,0 +1,10 @@
...
+    .max(26, { message: 'The value must be maximum 26 numbers long' }),
```
Comment (krzkaczor) [inline misc/unitSchema.ts:9]:
```suggestion
    .max(26, { message: 'The value must be less or equal to 26' }),
```

**Comment by krzkaczor** [comment]
I get an error at my face when I open token unit calc :( There should be some a reasonable defaults. 

image

**Review by krzkaczor — CHANGES_REQUESTED** [review]

**Review by krzkaczor — APPROVED** [review]

**Pull request merged**